projects/kit/components/slider/slider.component.ts

```diff
@@ -67,10 +69,10 @@ export class TuiSliderComponent {
     }
 
     get value(): number {
-        const {elementRef, control} = this;
-        const noKeySteps = !elementRef.nativeElement.hasAttribute(`data-key-steps`);
+        const {elementRef, control, injector} = this;
+        const hasKeySteps = Boolean(injector.get(TuiSliderKeyStepsDirective, null));
```

Comment:
We should probably memoize it, right now we rely on `get` operation of the `Injector` being getter-level fast which we cannot be certain of.

Reply:
fixed